plutus-tx/src/PlutusTx/Builtins/Internal.hs

```diff
@@ -147,15 +154,57 @@ equalsInteger = coerce ((==) @Integer)
 BYTESTRING
 -}
 
+{- Note [Silly sum types]
+
+GHC's worker/wrapper transformation and CPR analysis really like to tear
+product types apart. Usually, that's a good thing. But for our purposes, it's
+not. We want to see the actual `BuiltInByteString` or `BuiltInString`, not its
```

Comment:
case: Builtin

Reply:
I'll fix it, but it looks like this PR may be in a really backed up CI pipeline, and I'd rather not interrupt it. No idea why it's been pending so many hours.